asmcomp/arm64/emit.mlp

```diff
@@ -518,10 +518,18 @@ let assembly_code_for_allocation ?label_after_call_gc i ~n ~far =
   if !fastcode_flag then begin
     let lbl_redo = new_label() in
     let lbl_call_gc = new_label() in
-    `{emit_label lbl_redo}:`;
-    `	sub	{emit_reg reg_alloc_ptr}, {emit_reg reg_alloc_ptr}, #{emit_int n}\n`;
-    `	cmp	{emit_reg reg_alloc_ptr}, {emit_reg reg_alloc_limit}\n`;
-    `	add	{emit_reg i.res.(0)}, {emit_reg reg_alloc_ptr}, #8\n`;
+    if n > 4096 then begin
```

Comment:
Hi @xavierleroy,

A quick experiment shows that 4096 is encodable with the shifting form of the instruction.
The assembler automatically selects this.


cat t.s
	sub x0, x0, #4096
swalk@arm64:~$ cc -c t.s
swalk@arm64:~$ objdump -d t.o
0000000000000000 <.text>:
   0:	d1400400 	sub	x0, x0, #0x1, lsl #12

 cat t1.s
	sub x0, x0, #4097
swalk@arm64:~$ cc -c t1.s
t1.s: Assembler messages:
t1.s:1: Error: immediate out of range